remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full

bdhirsh opened the pull request:
Now that all ops are c10-full, we don't need the specialization of BoxedKernelWrapper that assumes multiple `out` arguments at the beginning of the signature. We can't remove the single `out` argument legacy specialization though, since it's also used for `in-place` ops.

Mostly put this up in the hopes that it would fix the windows failure in my dispatch key-plumbing PR, but it doesn't look like it did.

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #51363 update autograd kernels to use redispatch
* #51474 codegen'd API for redispatching
* #50848 inlining tensor.device()
* #50891 skip a second call to shouldUseRecordFunction for BackendSelect ops
* #49354 Plumbing TLS keys through the dispatcher
* **#50963 remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full**

Differential Revision: [D26026665](https://our.internmc.facebook.com/intern/diff/D26026665)

bdhirsh committed:
remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full

[ghstack-poisoned]

bdhirsh committed:
Update on "remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full"


Differential Revision: [D26026665](https://our.internmc.facebook.com/intern/diff/D26026665)

[ghstack-poisoned]

bdhirsh committed:
Update on "remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full"

Now that all ops are c10-full, we don't need the specialization of BoxedKernelWrapper that assumes multiple `out` arguments at the beginning of the signature. We can't remove the single `out` argument legacy specialization though, since it's also used for `in-place` ops.

Mostly put this up in the hopes that it would fix the windows failure in my dispatch key-plumbing PR, but it doesn't look like it did.


Differential Revision: [D26026665](https://our.internmc.facebook.com/intern/diff/D26026665)

[ghstack-poisoned]

bhosmer commented on aten/src/ATen/core/boxing/KernelFunction_test.cpp:
```diff
@@ -143,30 +143,6 @@ void boxed_func_for_outofplace_op(const OperatorHandle& /*opHandle*/, Stack* sta
 }
 
 void boxed_func_for_outofplace_multi_op(const OperatorHandle& /*opHandle*/, Stack* stack) {
-  // (Tensor(a!), Tensor(b!), Scalar, Scalar) -> (Tensor(a!), Tensor(b!))
```
oof, sorry about the naming dyslexia here and below re "legacy"

bhosmer reviewed (approved):
So nice to see this simplification - thanks for doing it!

bdhirsh committed:
Update on "remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full"

Now that all ops are c10-full, we don't need the specialization of BoxedKernelWrapper that assumes multiple `out` arguments at the beginning of the signature. We can't remove the single `out` argument legacy specialization though, since it's also used for `in-place` ops.

Mostly put this up in the hopes that it would fix the windows failure in my dispatch key-plumbing PR, but it doesn't look like it did.


Differential Revision: [D26026665](https://our.internmc.facebook.com/intern/diff/D26026665)

[ghstack-poisoned]

bdhirsh committed:
Update on "remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full"

Now that all ops are c10-full, we don't need the specialization of BoxedKernelWrapper that assumes multiple `out` arguments at the beginning of the signature. We can't remove the single `out` argument legacy specialization though, since it's also used for `in-place` ops.

Mostly put this up in the hopes that it would fix the windows failure in my dispatch key-plumbing PR, but it doesn't look like it did.


Differential Revision: [D26026665](https://our.internmc.facebook.com/intern/diff/D26026665)

[ghstack-poisoned]

bdhirsh committed:
Update on "remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full"

Now that all ops are c10-full, we don't need the specialization of BoxedKernelWrapper that assumes multiple `out` arguments at the beginning of the signature. We can't remove the single `out` argument legacy specialization though, since it's also used for `in-place` ops.

Mostly put this up in the hopes that it would fix the windows failure in my dispatch key-plumbing PR, but it doesn't look like it did.


Differential Revision: [D26026665](https://our.internmc.facebook.com/intern/diff/D26026665)

[ghstack-poisoned]

bdhirsh committed:
Update on "remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full"

Now that all ops are c10-full, we don't need the specialization of BoxedKernelWrapper that assumes multiple `out` arguments at the beginning of the signature. We can't remove the single `out` argument legacy specialization though, since it's also used for `in-place` ops.

Mostly put this up in the hopes that it would fix the windows failure in my dispatch key-plumbing PR, but it doesn't look like it did.


Differential Revision: [D26026665](https://our.internmc.facebook.com/intern/diff/D26026665)

[ghstack-poisoned]

bdhirsh committed:
Update on "remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full"

Now that all ops are c10-full, we don't need the specialization of BoxedKernelWrapper that assumes multiple `out` arguments at the beginning of the signature. We can't remove the single `out` argument legacy specialization though, since it's also used for `in-place` ops.

Mostly put this up in the hopes that it would fix the windows failure in my dispatch key-plumbing PR, but it doesn't look like it did.


Differential Revision: [D26026665](https://our.internmc.facebook.com/intern/diff/D26026665)

[ghstack-poisoned]

bdhirsh committed:
Update on "remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full"

Now that all ops are c10-full, we don't need the specialization of BoxedKernelWrapper that assumes multiple `out` arguments at the beginning of the signature. We can't remove the single `out` argument legacy specialization though, since it's also used for `in-place` ops.

Mostly put this up in the hopes that it would fix the windows failure in my dispatch key-plumbing PR, but it doesn't look like it did.


Differential Revision: [D26026665](https://our.internmc.facebook.com/intern/diff/D26026665)

[ghstack-poisoned]

bdhirsh committed:
Update on "remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full"

Now that all ops are c10-full, we don't need the specialization of BoxedKernelWrapper that assumes multiple `out` arguments at the beginning of the signature. We can't remove the single `out` argument legacy specialization though, since it's also used for `in-place` ops.

Mostly put this up in the hopes that it would fix the windows failure in my dispatch key-plumbing PR, but it doesn't look like it did.


Differential Revision: [D26026665](https://our.internmc.facebook.com/intern/diff/D26026665)

[ghstack-poisoned]

bdhirsh committed:
Update on "remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full"

Now that all ops are c10-full, we don't need the specialization of BoxedKernelWrapper that assumes multiple `out` arguments at the beginning of the signature. We can't remove the single `out` argument legacy specialization though, since it's also used for `in-place` ops.

Mostly put this up in the hopes that it would fix the windows failure in my dispatch key-plumbing PR, but it doesn't look like it did.


Differential Revision: [D26026665](https://our.internmc.facebook.com/intern/diff/D26026665)

[ghstack-poisoned]

bdhirsh committed:
Update on "remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full"

Now that all ops are c10-full, we don't need the specialization of BoxedKernelWrapper that assumes multiple `out` arguments at the beginning of the signature. We can't remove the single `out` argument legacy specialization though, since it's also used for `in-place` ops.

Mostly put this up in the hopes that it would fix the windows failure in my dispatch key-plumbing PR, but it doesn't look like it did.


Differential Revision: [D26026665](https://our.internmc.facebook.com/intern/diff/D26026665)

[ghstack-poisoned]

bdhirsh committed:
Update on "remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full"

Now that all ops are c10-full, we don't need the specialization of BoxedKernelWrapper that assumes multiple `out` arguments at the beginning of the signature. We can't remove the single `out` argument legacy specialization though, since it's also used for `in-place` ops.

Mostly put this up in the hopes that it would fix the windows failure in my dispatch key-plumbing PR, but it doesn't look like it did.


Differential Revision: [D26026665](https://our.internmc.facebook.com/intern/diff/D26026665)

[ghstack-poisoned]

bdhirsh committed:
Update on "remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full"

Now that all ops are c10-full, we don't need the specialization of BoxedKernelWrapper that assumes multiple `out` arguments at the beginning of the signature. We can't remove the single `out` argument legacy specialization though, since it's also used for `in-place` ops.

Mostly put this up in the hopes that it would fix the windows failure in my dispatch key-plumbing PR, but it doesn't look like it did.


Differential Revision: [D26026665](https://our.internmc.facebook.com/intern/diff/D26026665)

[ghstack-poisoned]

bdhirsh committed:
Update on "remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full"

Now that all ops are c10-full, we don't need the specialization of BoxedKernelWrapper that assumes multiple `out` arguments at the beginning of the signature. We can't remove the single `out` argument legacy specialization though, since it's also used for `in-place` ops.

Mostly put this up in the hopes that it would fix the windows failure in my dispatch key-plumbing PR, but it doesn't look like it did.


Differential Revision: [D26026665](https://our.internmc.facebook.com/intern/diff/D26026665)

[ghstack-poisoned]

bdhirsh committed:
Update on "remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full"

Now that all ops are c10-full, we don't need the specialization of BoxedKernelWrapper that assumes multiple `out` arguments at the beginning of the signature. We can't remove the single `out` argument legacy specialization though, since it's also used for `in-place` ops.

Mostly put this up in the hopes that it would fix the windows failure in my dispatch key-plumbing PR, but it doesn't look like it did.


Differential Revision: [D26026665](https://our.internmc.facebook.com/intern/diff/D26026665)

[ghstack-poisoned]

bdhirsh committed:
Update on "remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full"

Now that all ops are c10-full, we don't need the specialization of BoxedKernelWrapper that assumes multiple `out` arguments at the beginning of the signature. We can't remove the single `out` argument legacy specialization though, since it's also used for `in-place` ops.

Mostly put this up in the hopes that it would fix the windows failure in my dispatch key-plumbing PR, but it doesn't look like it did.


Differential Revision: [D26026665](https://our.internmc.facebook.com/intern/diff/D26026665)

[ghstack-poisoned]

bdhirsh committed:
Update on "remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full"

Now that all ops are c10-full, we don't need the specialization of BoxedKernelWrapper that assumes multiple `out` arguments at the beginning of the signature. We can't remove the single `out` argument legacy specialization though, since it's also used for `in-place` ops.

Mostly put this up in the hopes that it would fix the windows failure in my dispatch key-plumbing PR, but it doesn't look like it did.


Differential Revision: [D26026665](https://our.internmc.facebook.com/intern/diff/D26026665)

[ghstack-poisoned]

bdhirsh committed:
Update on "remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full"

Now that all ops are c10-full, we don't need the specialization of BoxedKernelWrapper that assumes multiple `out` arguments at the beginning of the signature. We can't remove the single `out` argument legacy specialization though, since it's also used for `in-place` ops.

Mostly put this up in the hopes that it would fix the windows failure in my dispatch key-plumbing PR, but it doesn't look like it did.


Differential Revision: [D26026665](https://our.internmc.facebook.com/intern/diff/D26026665)

[ghstack-poisoned]

bdhirsh committed:
Update on "remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full"

Now that all ops are c10-full, we don't need the specialization of BoxedKernelWrapper that assumes multiple `out` arguments at the beginning of the signature. We can't remove the single `out` argument legacy specialization though, since it's also used for `in-place` ops.

Mostly put this up in the hopes that it would fix the windows failure in my dispatch key-plumbing PR, but it doesn't look like it did.


Differential Revision: [D26026665](https://our.internmc.facebook.com/intern/diff/D26026665)

[ghstack-poisoned]

bdhirsh committed:
Update on "remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full"

Now that all ops are c10-full, we don't need the specialization of BoxedKernelWrapper that assumes multiple `out` arguments at the beginning of the signature. We can't remove the single `out` argument legacy specialization though, since it's also used for `in-place` ops.

Mostly put this up in the hopes that it would fix the windows failure in my dispatch key-plumbing PR, but it doesn't look like it did.


Differential Revision: [D26026665](https://our.internmc.facebook.com/intern/diff/D26026665)

[ghstack-poisoned]

bdhirsh committed:
Update on "remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full"

Now that all ops are c10-full, we don't need the specialization of BoxedKernelWrapper that assumes multiple `out` arguments at the beginning of the signature. We can't remove the single `out` argument legacy specialization though, since it's also used for `in-place` ops.

Mostly put this up in the hopes that it would fix the windows failure in my dispatch key-plumbing PR, but it doesn't look like it did.


Differential Revision: [D26026665](https://our.internmc.facebook.com/intern/diff/D26026665)

[ghstack-poisoned]

bdhirsh committed:
Update on "remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full"

Now that all ops are c10-full, we don't need the specialization of BoxedKernelWrapper that assumes multiple `out` arguments at the beginning of the signature. We can't remove the single `out` argument legacy specialization though, since it's also used for `in-place` ops.

Mostly put this up in the hopes that it would fix the windows failure in my dispatch key-plumbing PR, but it doesn't look like it did.


Differential Revision: [D26026665](https://our.internmc.facebook.com/intern/diff/D26026665)

[ghstack-poisoned]

bdhirsh committed:
Update on "remove unnecessary BoxedKernelWrapper specialization now that ops are all c10-full"

Now that all ops are c10-full, we don't need the specialization of BoxedKernelWrapper that assumes multiple `out` arguments at the beginning of the signature. We can't remove the single `out` argument legacy specialization though, since it's also used for `in-place` ops.

Mostly put this up in the hopes that it would fix the windows failure in my dispatch key-plumbing PR, but it doesn't look like it did.


Differential Revision: [D26026665](https://our.internmc.facebook.com/intern/diff/D26026665)

[ghstack-poisoned]
